blivet/populator.py

```diff
@@ -429,6 +429,13 @@ def addUdevPartitionDevice(self, info, disk=None):
                 lvm.lvm_cc_addFilterRejectRegexp(name)
                 return
 
+        # If this is the live device protect the whole disk
+        if self.liveBackingDevice == name and not disk.protected \
+           and not disk.name in self.protectedDevNames:
```

Comment:
I would prefer if you would switch from `not disk.name in` to `disk.name not in`, but otherwise this looks like a nice solution.